src/main/java/org/sagebionetworks/web/client/presenter/SettingsPresenter.java

```diff
@@ -333,4 +333,39 @@ public void invoke() {
 			}
 		});
 	}
+
+	@Override
+	public void changePassword() {
+		clearPasswordErrors();
+		String currentPassword = view.getCurrentPasswordField();
+		String password1 = view.getPassword1Field();
+		String password2 = view.getPassword2Field();
+		if (!checkPasswordDefined(currentPassword)) {
+			view.setCurrentPasswordErrorMessage(DisplayConstants.ERROR_ALL_FIELDS_REQUIRED);
+			view.setCurrentPasswordInError(true);
+		} else if (!checkPasswordDefined(password1)){
+			view.setPassword1ErrorMessage(DisplayConstants.ERROR_ALL_FIELDS_REQUIRED);
+			view.setPassword1InError(true);
+		} else if (!checkPasswordDefined(password2) || !password1.equals(password2)) {
```

Comment:
This is indeed the best solution, and I was really wanting some design overview regarding this issue, as the previous solution was pretty brute force.